### PR TITLE
doc: Improve scripted-diff developer docs

### PR DESCRIPTION
**Pull request by dongcarl**
Instead of verifying all scripted-diffs, provide an example that only verifies all scripted-diffs in commits since `origin/master`.

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -827,7 +827,13 @@ To create a scripted-diff:
     - `-BEGIN VERIFY SCRIPT-`
     - `-END VERIFY SCRIPT-`
 
-The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`
+The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`. The tool's default behavior when supplied
+with a commit is to verify all scripted-diffs since the beginning of time. For development, it might be more convienient
+to verify all scripted-diffs in commits since `origin/master`:
```
Comment (MarcoFalke) [inline doc/developer-notes.md]:
```suggestion
to verify all scripted-diffs in a range `A..B`, for example:
```

nit: Not sure if worth changing, but it should accept any range.

**Comment by MarcoFalke** [comment]
ACK 2734aaf1897b1cb6c0fd73d220980fd698e1b934

File: doc/developer-notes.md [hunk 2]
```diff
@@ -827,7 +827,16 @@ To create a scripted-diff:
     - `-BEGIN VERIFY SCRIPT-`
     - `-END VERIFY SCRIPT-`
 
-The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`
+The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`. The tool's default behavior when supplied
+with a commit is to verify all scripted-diffs from the beginning of time up to said commit. Internally, the tool passes
+the first supplied argument to `git rev-list --reverse` to determine which commits to verify script-diffs for, ignoring
+commits that don't conform to the commit message format described above.
+
+For development, it might be more convienient to verify all scripted-diffs in a range `A..B`, for example:
```
Comment (practicalswift) [inline doc/developer-notes.md]: Convenient :-)

**Comment by promag** [comment]
ACK 07e286d.

**Comment by practicalswift** [comment]
ACK 07e286d94038f52aa3969f8d3d8c6aeab8eb3d51

**Pull request merged**